### PR TITLE
Added check for gatsby in devDependencies.

### PR DESCRIPTION
**Pull request by michaek**
Following from discussion in #832, this adds a check for Gatsby in the current project's devDependencies in addition to dependencies.

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Thanks!

**Comment by systimotic** [comment]
Hi! I came across this PR and have a question about it. The checks added here look like they were already present in the `else if` statement below. Am I missing something?

File: packages/gatsby-cli/src/index.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ let inGatsbySite = false
 let localPackageJSON
 try {
   localPackageJSON = require(path.resolve(`./package.json`))
-  if (localPackageJSON.dependencies && localPackageJSON.dependencies.gatsby) {
+  if (localPackageJSON.dependencies && localPackageJSON.dependencies.gatsby || localPackageJSON.devDependencies && localPackageJSON.devDependencies.gatsby) {
     inGatsbySite = true
```
Comment (jquense) [inline packages/gatsby-cli/src/index.js]: I think this check is right below this one already?

**Comment by jquense** [comment]
yeah it is...weird

**Comment by KyleAMathews** [comment]
Hmmm yeah that was a mistake.

@jquense I'll assume your cli refactor fixes this? :-)

**Comment by jquense** [comment]
it does!

**Comment by systimotic** [comment]
Sorry for bothering you about this then, thanks for looking at it! Thanks for your work on Gatsby as well, I'm loving it! ❤️

**Comment by jlengstorf** [comment]
Hiya @michaek! 👋

This is definitely late, but on behalf of the entire Gatsby community, I wanted to say thank you for being here.

Gatsby is built by awesome people like you. Let us say “thanks” in two ways:

1.  **We’d like to send you some Gatsby swag.** As a token of our appreciation, you can go to the [Gatsby Swag Store][store] and log in with your GitHub account to get a coupon code good for one free piece of swag. (We’ve got t-shirts and hats, plus some socks that are really razzing our berries right now.)
2.  **If you’re not already part of it, we just invited you to join the Gatsby organization on GitHub.** This will add you to our team of maintainers. You’ll receive an email shortly asking you to confirm. By joining the team, you’ll be able to label issues, review pull requests, and merge approved pull requests.

If you have questions, please don’t hesitate to reach out to us: tweet at [@gatsbyjs][twitter] and we’ll come a-runnin’.

Thanks again! 💪💜

[store]: https://store.gatsbyjs.org
[twitter]: https://twitter.com/gatsbyjs

**Pull request merged**